diff --git a/lib/feature.rb b/lib/feature.rb
index f8d34e9c386..549c7fc063e 100644
--- a/lib/feature.rb
+++ b/lib/feature.rb
@@ -37,6 +37,7 @@ def get(key)
 
     def persisted_names
       return [] unless Gitlab::Database.main.exists?
+      return [] unless Feature::FlipperFeature.table_exists?
 
       # This loads names of all stored feature flags
       # and returns a stable Set in the following order:
@@ -74,6 +75,7 @@ def enabled?(key, thing = nil, type: :development, default_enabled: false)
       # During setup the database does not exist yet. So we haven't stored a value
       # for the feature yet and return the default.
       return default_enabled unless Gitlab::Database.main.exists?
+      return default_enabled unless Feature::FlipperFeature.table_exists?
 
       feature = get(key)
 
